documentation/context.tex

```diff
@@ -0,0 +1,222 @@
+	\item Array of \emph{spaces} that are present in the address. 
+	\item \emph{Offset} from the bases.
+\end{itemize}
+
+The common composition of an address is one base section (as the start of the address) and value of storage (as the offset from it).
+
+The need for the whole array of bases is because addresses from different sections can be arbitrarily added or subtracted. This information is needed as the correct sequence of arithmetic operations can reduce number of bases (even spaces) to zero and create absolute value. This value can be later used in places where a relocatable value would be forbidden.
+
+\paragraph*{Space} is block of bytes with yet not known length. It is created in the active location counter when execution of counter's operation can not be performed due to non previously defined ordinary symbols. See the different kind of spaces and the reason of creation in  the \cref{tab06:space}.
+
+\begin{table}
+	\centering
+	\begin{tabular}{lr}
+		\textbf{Space Kind} &                                          \textbf{Creation} \\ \toprule
+		Ordinary            &            when instruction outputs data of unknown length \\
+		LOCTR begin         &  when defining more than one location counter in a section \\
+		Alignment           &   when current alignment is unknown due to previous spaces \\
+		LOCTR set           &     when moving counter's value to the address with spaces \\
+		LOCTR max           & when moving counter's value to the next available location \\
+		LOCTR unknown       &     when moving counter's value to the yet unknown address \\ \bottomrule
+	\end{tabular}
+	\caption{Different kinds of spaces and reasons of creation.}
+	\label{tab06:space}
+\end{table}
+
+
+When a space length becomes known, all addresses containing the spaces need to be updated (remove the space and append offset). Therefore, space structure contains an array of address listeners. Hence, when an address is assigned a relocatable value that contains the space, the address is added to its array. This serves as an easy point of space resolving.
+
+\vspace{0.5cm}
+
+ORG instruction can arbitrarily move location counter's value forward and backward. With addition to that, ORG  can also order location counter to set it's value to the next available value (the lowest untouched address, see \cref{loctr}). Combining this with the possible spaces creation, location counter holds an array of the location counter data to properly set the next available value.
+
+\subsubsection{Symbol dependency tables}
+\label{symbol_dependency_tables}
+HLASM forbids cyclic symbol definition. This component maintains dependencies between symbols and detects possible cycles.
+Let us describe the main components of dependency resolving:
+
+\paragraph*{Dependant} is a structure used in the symbol dependency tables. It encapsulates objects that can be dependent on another. Dependant object can be a \emph{symbol}, \emph{symbol attribute} and \emph{space}.
```

Comment:
tu by to chcelo asi itemize, kedze je to uvedene dvojbodkou.

tiez: v niektorych tych paragrafoch sa k tomu v bolde spravas ako k nadpisu a zacinas novou vetou, v niektorych uz to prve slovo je sucastou vety, bolo by fajn zjednotit to.

Reply:
itemize by podla mna vizualne zmensilo dolezitost tych paragrafov, zmazem dvojbodku.